### PR TITLE
fix: Resolve issues raised by `linode_stackscript` migration

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This change resolves issues that arose from the `linode_stackscript` migration to Framework.

## ✔️ How to Test

```
make PKG_NAME=linode/stackscript testacc
```

### CODE REVIEW

File: linode/stackscript/framework_resource_schema.go [hunk 1]
```diff
@@ -51,6 +62,7 @@ var frameworkResourceSchema = schema.Schema{
 			Computed: true,
 			PlanModifiers: []planmodifier.Bool{
 				boolplanmodifier.RequiresReplace(),
```
[the remark this answers is not in the document]
Comment (lgarber-akamai) [reply linode/stackscript/framework_resource_schema.go]: Great idea!

File: linode/stackscript/framework_models.go [hunk 2]
```diff
@@ -32,17 +32,46 @@ type StackScriptModel struct {
...
+	// Only update the images return if there is a change
+	var plannedImages []string
+	diagnostics.Append(data.Images.ElementsAs(ctx, &plannedImages, false)...)
+	if diagnostics.HasError() {
+		return diagnostics
+	}
+
+	if !helper.StringListElementsEqual(plannedImages, stackscript.Images) {
+		remoteImages, err := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
+		if err != nil {
```
Comment (zliang-akamai) [inline linode/stackscript/framework_models.go:59]:
```suggestion
		if err.HasError() {
```

File: linode/stackscript/framework_models.go [hunk 3]
```diff
@@ -32,17 +32,46 @@ type StackScriptModel struct {
...
+}
+
+func (data *StackScriptModel) parseComputedAttributes(
+	ctx context.Context,
+	stackscript *linodego.Stackscript,
+) diag.Diagnostics {
+	data.ID = types.StringValue(strconv.Itoa(stackscript.ID))
+
 	images, err := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
 	if err != nil {
```
Comment (zliang-akamai) [inline linode/stackscript/framework_models.go:76]:
```suggestion
	if err.HasError() {
```

File: linode/stackscript/framework_models.go [hunk 4]
```diff
@@ -32,17 +32,46 @@ type StackScriptModel struct {
...
+	var plannedImages []string
+	diagnostics.Append(data.Images.ElementsAs(ctx, &plannedImages, false)...)
+	if diagnostics.HasError() {
+		return diagnostics
+	}
+
+	if !helper.StringListElementsEqual(plannedImages, stackscript.Images) {
+		remoteImages, err := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
+		if err != nil {
+			return err
```
Comment (zliang-akamai) [inline linode/stackscript/framework_models.go:60]:
```suggestion
		remoteImages, diags := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
		if diags.HasError() {
```

File: linode/stackscript/framework_models.go [hunk 5]
```diff
@@ -32,17 +32,46 @@ type StackScriptModel struct {
...
+
+func (data *StackScriptModel) parseComputedAttributes(
+	ctx context.Context,
+	stackscript *linodego.Stackscript,
+) diag.Diagnostics {
+	data.ID = types.StringValue(strconv.Itoa(stackscript.ID))
+
 	images, err := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
 	if err != nil {
 		return err
```
Comment (zliang-akamai) [inline linode/stackscript/framework_models.go:77]:
```suggestion
	images, diags := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
	if diags.HasError() {
		return diags
```

File: linode/stackscript/framework_models.go (hunk 4 above)
Comment (zliang-akamai) [reply linode/stackscript/framework_models.go:60]: There may be this type of thing elsewhere. Can you screen all `err != nil` to figure out if there should be `diags` instead?

**Comment by lgarber-akamai** [comment]
> Does `user_defined_fields` also need `UseStateForUnknown()`?

`user_defined_fields` does not since it's expected to change when the script updates 👍

**Review by zliang-akamai — APPROVED** [review]
Looks good to me!

**Review by zliang-akamai — APPROVED** [review]

**Comment by lgarber-akamai** [comment]
/acctest sha=90ee705da4aefe13887caa9134950dcda5fcbb1e pkg=linode/stackscript

**Review by ezilber-akamai — APPROVED** [review]
Oops, forgot to actually approve this before. Good work!

**Review by zliang-akamai — APPROVED** [review]

**Pull request merged**